src/config/config.rs

```diff
@@ -13,24 +13,55 @@ use log::warn;
 use std::{
     collections::HashMap,
     fs,
-    path::{Path, PathBuf},
+    path::{Component, Path, PathBuf},
     str::FromStr,
 };
 
+fn canonicalize(path: &Path) -> String {
+    let mut parts: Vec<String> = Vec::new();
+
+    for component in path.components() {
+        match component {
+            Component::RootDir | Component::CurDir => {}
+            Component::ParentDir => {
+                parts.pop();
+            }
+            c => parts.push(
+                c.as_os_str()
+                    .to_str()
+                    .expect("OsStr::to_str failed")
+                    .to_owned(),
+            ),
+        }
+    }
+    parts.join("/")
+}
+
 #[derive(Debug)]
 pub struct GirVersion {
-    pub gir_dir: PathBuf,
+    pub gir_dir: String,
+    file_name: Option<String>,
     hash: Option<String>,
 }
 
 impl GirVersion {
+    fn new(gir_dir: &Path, hash: Option<String>) -> Self {
+        Self {
+            gir_dir: canonicalize(gir_dir),
+            file_name: gir_dir
+                .file_name()
+                .map(|s| s.to_str().expect("OsStr::to_str failed").to_owned()),
+            hash,
+        }
+    }
+
     pub fn get_hash(&self) -> &str {
         self.hash.as_deref().unwrap_or("???")
     }
 
     pub fn get_repository_url(&self) -> Option<&str> {
-        match self.gir_dir.file_name() {
-            Some(r) if r == "gir-files" => Some("https://github.com/gtk-rs/gir-files"),
+        match self.file_name {
+            Some(ref r) if r == "gir-files" => Some("https://github.com/gtk-rs/gir-files"),
```

Comment:
Why hardcode the gtk-rs gir-files URL here?

Reply:
It was done like this, keeping the same logic. Can be removed in the other PR.